### PR TITLE
add String() method to Decimal

### PR DESCRIPTION
**Pull request by veqryn**
Right now, `Decimal` only has a `Float64` method and an unused `toString()`, which returns just the data type.

The Float64 method is great, but like all floats, it can introduce errors because of how floats work.

Example:
```golang
	// This outputs: 0.008216620000000001
	// It should be: 0.00821662
	fmt.Println((&duckdb.Decimal{
		Width: 18,
		Scale: 8,
		Value: big.NewInt(821662),
	}).Float64())
```

This PR adds a Stinger method allowing a lossless and correct conversion from Decimal to a string.

### CODE REVIEW

File: types.go [hunk 1]
```diff
@@ -130,6 +131,36 @@ func (d *Decimal) Float64() float64 {
...
+	// Pad a number with 0.0's if needed
+	if len(zeroTrimmed) <= scale {
+		return fmt.Sprintf("%s0.%s%s", signStr, strings.Repeat("0", scale-len(zeroTrimmed)), zeroTrimmed)
+	}
+	return signStr + zeroTrimmed[:len(zeroTrimmed)-scale] + "." + zeroTrimmed[len(zeroTrimmed)-scale:]
+}
+
 func (d *Decimal) toString() string {
 	return fmt.Sprintf("DECIMAL(%d,%d)", d.Width, d.Scale)
 }
```
Comment (taniabogatsch) [inline types.go:166]: Let's remove this. As you pointed out, it is not exported and is unused. 😅

File: types_test.go [hunk 2]
```diff
@@ -381,6 +381,151 @@ func TestDecimal(t *testing.T) {
...
+			},
+			expected: "0.00821662",
+		},
+	}
+
+	for _, tc := range testCases {
+		actual := tc.input.String()
+		if actual != tc.expected {
+			t.Errorf("ToString(%v) => %q, want %q", tc.input, actual, tc.expected)
+		}
```
Comment (taniabogatsch) [inline types_test.go:525]: For consistency with our other tests:
```Go
require.Equal(t, tc.expected, actual)
```

File: types.go (hunk 1 above)
Comment (veqryn) [reply types.go:166]: 👍

File: types_test.go (hunk 2 above)
Comment (veqryn) [reply types_test.go:525]: 👍

**Comment by veqryn** [comment]
Thank you @taniabogatsch , should be good now

**Comment by taniabogatsch** [comment]
Thanks!

**Pull request merged**